src/ray/core_worker/core_worker.cc

```diff
@@ -1757,13 +1750,14 @@ void CoreWorker::HandleDirectActorCallArgWaitComplete(
 void CoreWorker::HandleGetObjectStatus(const rpc::GetObjectStatusRequest &request,
                                        rpc::GetObjectStatusReply *reply,
                                        rpc::SendReplyCallback send_reply_callback) {
+  if (HandleWrongRecipient(WorkerID::FromBinary(request.owner_worker_id()),
```

Comment:
Is this behavior change intended? Handle wrong recipient will return an error status instead of OK.

Reply:
Oh yeah, that should be fine. The client does the same thing regardless of the return status.

Reply:
(It just logs an error on a bad return status)

Reply:
We don't want to log the error right, since this is a normal path. How about keeping the original handler and adding a comment?

Reply:
Hmm this isn't really a normal path, though. It means that the original owner died. You would get the same error message with a different order of events where the owner dies after the `GetObjectStatus` request gets sent.

Reply:
The comment says it's for handling status for objects produced by a previous task. So it might be multiple tasks run on one worker right?

Reply:
Oh I see, this is no longer checking the task ID but only the worker ID. Can you remove the LOG(INFO) below then?